criu/seize.c

```diff
@@ -517,17 +518,22 @@ static int check_freezer_cgroup(void)
 		return -1;
 
 	state = get_freezer_state(fd);
-	close(fd);
 	if (state == FREEZER_ERROR) {
-		return -1;
+		goto err;
 	}
 
+	origin_freezer_state = state == FREEZING ? FROZEN : state;
```

Comment:
freeze_processes() initializes the origin_freezer_state again, but we have to be sure that it contains the state that has been initially.

Reply:
> freeze_processes() initializes the origin_freezer_state again

When the CUDA plugin is used and `freeze_cgroup_disabled` is set to `true`, CRIU calls `check_freezer_cgroup()` instead of `freeze_processes()`:
```C
	if (opts.freeze_cgroup && !freeze_cgroup_disabled) {
		if (freeze_processes())
			goto err;
	} else {
		if (opts.freeze_cgroup && check_freezer_cgroup())
			goto err;
		if (compel_interrupt_task(pid)) {
			set_cr_errno(ESRCH);
			goto err;
		}
	}

```

Reply:
It is even more sophisticated. Pls rename freeze_cgroup_disabled and check_freezer_cgroup to something more suitable for the new reality.

Reply:
Thank you Andrei! I've updated the pull request to rename `freeze_cgroup_disabled` to `compel_interrupt_only_mode` and `check_freezer_cgroup` to `prepare_freezer_for_interrupt_only_mode`.